File: programming files/remove_duplicates.py
# Removed sort at the beginning. Mess up the duplicate check (== instead of !=). : NSH

'''
Function that sorts a list and removes duplicates
'''
def remove_duplicates(arr):
    unique_arr = []
    removed_elements = []
    # Check if each element if unique, otherwise add it to a list of removed elements
    for i in range(len(arr)):
        if i == 0 or arr[i] == arr[i-1]:
            unique_arr.append(arr[i])
        else:
            removed_elements.append(arr[i])
    return unique_arr, removed_elements

# Test the function
arr = [1, 2, 2, 3, 4, 4, 5]
# Print the original list, the list with duplicates removed, and the removed elements
unique_arr, removed_elements = remove_duplicates(arr)
print("Original array:", arr)
print("Unique array:", unique_arr)
print("Removed elements:", removed_elements)

File: programming files/remove_duplicates.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class remove_duplicates {

    // Function to remove duplicates from the input list 'arr' and keep track of removed elements
    public static List<Integer> removeDuplicates(List<Integer> arr, List<Integer> removedElements) {
        
        // Create a new list to store unique elements
        List<Integer> uniqueArr = new ArrayList<>();

        // Iterate through the input list 'arr'
        for (int i = 0; i < arr.size(); i++) {

            // Check if 'uniqueArr' does not contain the current element
            if (!uniqueArr.contains(arr.get(i))) {

                // If not contained, add it to 'uniqueArr'
                uniqueArr.add(arr.get(i));
            } else {
                // If already contained, add it to the 'removedElements' list
                removedElements.add(arr.get(i));
            }
        }
        // Return the list of unique elements
        return uniqueArr;
    }

    public static void main(String[] args) {
        // Test the function
        List<Integer> arr = Arrays.asList(1, 2, 2, 3, 4, 4, 5);
        List<Integer> removedElements = new ArrayList<>();
        List<Integer> uniqueArr = removeDuplicates(arr, removedElements);
        
        // Print the original array, unique array, and removed elements
        System.out.println("Original array: " + arr);
        System.out.println("Unique array: " + uniqueArr);
        System.out.println("Removed elements: " + removedElements);
    }
}
